src/nvme/nbft.h

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+/*
+ * This file is part of libnvme.
+ * Copyright (c) 2021-2022, Dell Inc. or its subsidiaries.  All Rights Reserved.
+ *
+ * Authors: Stuart Hayes <Stuart_Hayes@Dell.com>
+ *
+ */
+#ifndef _NBFT_H
+#define _NBFT_H
+
+#include <uuid/uuid.h>
+#include <ccan/list/list.h>
+
+enum primary_admin_host_flag {
+	not_indicated,
+	unselected,
+	selected,
+	reserved,
+};
+
+struct nbft_host {
+	uuid_t *id;
+	char *nqn;
+	bool host_id_configured;
+	bool host_nqn_configured;
+	enum primary_admin_host_flag primary;
+};
+
+struct nbft_hfi_info_tcp {
+	__u32 pci_sbdf;
+	__u8 *mac_addr;
+	__u16 vlan;
+	__u8 ip_origin;
+	char ipaddr[40];
+	__u8 subnet_mask_prefix;
+	char gateway_ipaddr[40];
+	__u16 route_metric;
+	char primary_dns_ipaddr[40];
+	char secondary_dns_ipaddr[40];
+	char dhcp_server_ipaddr[40];
+	char *host_name;
+	bool this_hfi_is_default_route;
+	bool dhcp_override;
+};
+
+struct nbft_hfi {
+	int index;
+	char transport[8];
+	//uuid_t *host_id;
+	//char *host_nqn;
+	struct nbft_hfi_info_tcp tcp_info;
+	struct list_node node;
+};
+
+struct nbft_discovery {
+	int index;
+	struct list_node node;
+	struct nbft_security *security;
+	struct nbft_hfi *hfi;
+	char *uri;
+	char *nqn;
+};
+
+struct nbft_security {
+	int index;
+	struct list_node node;
+	/* TODO add fields */
```

Comment:
Please add those if they are missing

Reply:
Postponed - see the corresponding `nbft.c` note below.